lib/resque/worker.rb

```diff
@@ -394,11 +399,13 @@ def register_signal_handlers
 
     def unregister_signal_handlers
       trap('TERM') do
-        trap ('TERM') do
-          # ignore subsequent terms
+        trap('TERM') do
+          # Ignore subsequent term signals
         end
+
         raise TermException.new("SIGTERM")
```

Comment:
@chrisccerami This `TermException` seems to be causing issues in my case. If `fork_per_job? && term_child` then this code is run on the child process which causes the child process to immediately exit with a `TermException` without waiting for the second `SIGTERM` initiated by the parent after `RESQUE_PRE_SHUTDOWN_TIMEOUT`